### PR TITLE
Rya 106 - Ensure users are able to access all or none of a Rya instance's tables.

### PR DESCRIPTION
**Pull request by kchilton2**
Added two new commands to the RyaClient class. AddUser and RemoveUser. Users that have been added to the Rya instance are tracked within the RyaDetails for the instance and they are granted full table permissions. When you revoke a user from the instance, they also have all table permissions revoked.

### CODE REVIEW

File: extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    public void removeUser(final String instanceName, final String username) throws InstanceDoesNotExistException, RyaClientException {
+        requireNonNull(instanceName);
+        requireNonNull(username);
+
+        // Update the instance details.
+        final RyaDetailsUpdater updater = new RyaDetailsUpdater( new AccumuloRyaInstanceDetailsRepository(getConnector(), instanceName) );
+        try {
+            updater.update(originalDetails -> RyaDetails.builder( originalDetails ).removeUser(username).build());
+        } catch (RyaDetailsRepositoryException | CouldNotApplyMutationException e) {
+            throw new RyaClientException("Could not add the user '" + username + "' to the Rya instance's details.", e);
```
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java]: Remove, not all.

File: extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        try {
+            // Build the list of tables that are present within the Rya instance.
+            final List<String> tables = new RyaTableNames().getTableNames(instanceName, getConnector());
+
+            // Update the user permissions for those tables.
+            for(final String table : tables) {
+                try {
+                    TABLE_PERMISSIONS.revokeAllPermissions(username, table, getConnector());
+                } catch (AccumuloException | AccumuloSecurityException e) {
+                    throw new RyaClientException("Could not grant access to table '" + table + "' for user '" + username + "'.", e);
```
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java]: Revoke, not grant.

File: extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java (hunk 1 above)
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java]: Done.

File: extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java (hunk 2 above)
Comment (kchilton2) [inline extras/indexing/src/main/java/mvm/rya/api/client/accumulo/AccumuloRemoveUser.java]: Done.

**Comment by pujav65** [comment]
looks good

**Comment by pujav65** [comment]
hey can you resolve the merge conflict?  I will merge after.

**Comment by isper3at** [comment]
This pull request is updated and clean

**Comment by amihalik** [comment]
Wow!  Thanks @isper3at 
I'll put this in the queue of things to merge after the release is finalized.

**Pull request closed without merging**